pkg/middleware/middleware.go

```diff
@@ -254,19 +253,7 @@ func WriteSessionCookie(ctx *models.ReqContext, value string, maxLifetimeDays in
 	}
 
 	ctx.Resp.Header().Del("Set-Cookie")
-	cookie := http.Cookie{
-		Name:     setting.LoginCookieName,
-		Value:    url.QueryEscape(value),
-		HttpOnly: true,
-		Path:     setting.AppSubUrl + "/",
-		Secure:   setting.CookieSecure,
-		MaxAge:   maxAge,
-	}
-	if setting.CookieSameSite != http.SameSiteDefaultMode {
-		cookie.SameSite = setting.CookieSameSite
-	}
-
-	http.SetCookie(ctx.Resp, &cookie)
+	WriteCookie(ctx.Resp, setting.LoginCookieName, url.QueryEscape(value), maxAge, nil)
```

Comment:
This must include proper config as last argument. Know it's tricky since you only have `setting.Cfg` available in some handlers and some not. Maybe you can add override functions in middleware/cookie.go to either use `setting.Cfg` or global `setting` that we eventually want to move away from.